### DISCUSSION TITLE
How to allow Fief admin to access restricted external database?

### DISCUSSION THREAD

**Discussion by johanhermansson** [fief-dev/fief#4] [Q&A] [answered]
We currently use DigitalOcean database. The database is restricted to specific ip addresses. Is it possible to allow an ip address/range to get it to work?

Not sure if it's possible to give a specific DB user permissions, so maybe that is also a caveat. 

Maybe we need to use your DB hosting, or self-hosting.

Any suggestions? :)

**Comment by frankie567** [comment]
Welcome to Fief, @johanhermansson! You're our first community member 😄

Ha, I was sure this kind of problems would pop-up quickly 😅 

Yes, the best thing to do would be to allow Fief's IP on your database server and that should work.

Problem is: my current hosting, Fly.io, uses VM on AWS and their internal IP's are highly dynamic 🥲 So I can't give you a precise and stable list for you to allow... Maybe I should question this and host Fief on a more traditional/static server.

In the meantime, I suggest you to try either the cloud database or the [self-hosting](https://docs.fief.dev/self-hosting/quickstart/) option.

**Reply by johanhermansson** [reply]
Understandable. Will probably check out the cloud database 👍 

I just thought that this topic needed a discussions if someone else want to know. Maybe it needs to be added to the docs? 🤓

**Reply by frankie567** [reply]
Thank you, you did very well to pop-up this subject! My goal is to make "Bring Your Own Database" our flagship feature, so everything that could help improve it is very precious.

I'll mention this in the docs 👍

**Comment by frankie567** [comment, marked as answer]
Hi @johanhermansson!

Just an update to tell you that I've changed our cloud hosting for a provider that can provide static IPs for you to allow list 🎉  Here they are:

```
18.156.158.53
18.156.42.200
52.59.103.54
```

If you have the chance, it would be great if you could try it with your MySQL database and tell me if it works!

**Reply by johanhermansson** [reply]
Connection works 🥳
